index.html

```diff
@@ -82,14 +82,42 @@
         app.appendChild(cards);
 
         function setTitle(title) {
+            const titleContainer = document.createElement("div");
+            titleContainer.style.display = "flex";
+            titleContainer.style.alignItems = "center";
+            titleContainer.style.justifyContent = "center";
+            titleContainer.style.marginBottom = "20px";
+            app.appendChild(titleContainer);
+
             document.title = title;
             const titleElement = document.createElement("h1");
             titleElement.textContent = title;
-            titleElement.style.marginBottom = "20px";
-            app.appendChild(titleElement);
+            titleElement.style.marginRight = "10px";
+            titleContainer.appendChild(titleElement);
+
+            const editTitleButton = document.createElement("button");
+            editTitleButton.classList.add("btn", "waves-effect", "waves-light");
+            titleContainer.appendChild(editTitleButton);
+
+            const editTitleIcon = document.createElement("i");
+            editTitleIcon.classList.add("material-icons");
+            editTitleIcon.textContent = "edit";
+            editTitleButton.appendChild(editTitleIcon);
+
+            editTitleButton.addEventListener("click", () => {
+                const newTitle = prompt("Edit title:", title);
+                if (newTitle !== null && newTitle.trim() !== "") {
+                    title = newTitle;
+                    document.title = title;
+                    titleElement.textContent = title;
+                    timers.title = title;
+                    updateUrlWithTimers(timers);
+                }
+            });
```

Comment:
Replace the existing setTitle() function in your code with this updated version. This implementation will use a Material Icon for the "Edit" button next to the title.